wrapper/src/main/java/software/amazon/jdbc/ds/AwsWrapperDataSource.java

```diff
@@ -102,11 +103,13 @@ public Connection getConnection(String username, String password) throws SQLExce
       if (!StringUtils.isNullOrEmpty(this.jdbcUrl)
           || (!StringUtils.isNullOrEmpty(this.urlPropertyName)
           && !StringUtils.isNullOrEmpty(props.getProperty(this.urlPropertyName)))) {
+        final Properties parsedProperties = new Properties();
         if (!StringUtils.isNullOrEmpty(this.jdbcUrl)) {
-          parsePropertiesFromUrl(this.jdbcUrl, props);
+          parsePropertiesFromUrl(this.jdbcUrl, parsedProperties);
         } else {
-          parsePropertiesFromUrl(props.getProperty(this.urlPropertyName), props);
+          parsePropertiesFromUrl(props.getProperty(this.urlPropertyName), parsedProperties);
         }
+        props.forEach(parsedProperties::putIfAbsent);
```

Comment:
I'm confused, it looks like parsedProperties is never used. We store the properties from the URL in it, then we store the properties from props in it (if absent), but we don't do anything with it after. Is this line supposed to be flipped  (`parsedProperties.forEach(props::putIfAbsent)`)?

Reply:
Good catch, thanks